### PR TITLE
Bug/6763 web data stream not available

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue:

- #6763

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics/components/settings/SettingsForm.stories.js
Comment (techanvil) [inline assets/js/modules/analytics/components/settings/SettingsForm.stories.js:284]: This story had degenerated into a broken state, and was fixed as a byproduct of this PR. This scenario was therefore added to provide some additional test coverage.

File: assets/js/modules/analytics/components/settings/GA4SettingsControls.js
Comment (kuasha420) [inline assets/js/modules/analytics/components/settings/GA4SettingsControls.js:235]: Unrelated refactor that was needed to satisfy the cyclometic complexity eslint rules. :sweat_smile:

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good, very minor copy edit change and a thought about the variable/selector naming here.

After that should be good to merge 👍🏻

File: assets/js/modules/analytics/components/settings/GA4SettingsControls.js
Comment (techanvil) [reply assets/js/modules/analytics/components/settings/GA4SettingsControls.js:235]: Just chipping in here to say, I think disabling the complexity rule here would be the wrong approach - it would be necessary to disable it for the whole function, meaning this function could continue to grow more complex with no limit to it. It seems a bit of a slippery slope to start disabling the rule like this.

If this refactor is objectionable, I'd suggest taking a different approach to refactoring the function rather than disabling the rule entirely...

Comment (kuasha420) [reply assets/js/modules/analytics/components/settings/GA4SettingsControls.js:235]: @techanvil @tofumatt yep, disabling the rule is per function and not per line so disabling the rule will really bite us in the future. I can try to find other way to claim back another execution path without this specific refactor, but as far as I remember this was the most straightforward and reasonable one :sweat_smile:

Comment (techanvil) [reply assets/js/modules/analytics/components/settings/GA4SettingsControls.js:235]: I actually don't mind the ternary here, but @tofumatt does seem quite strongly against it and I don't have a strong enough opinion on the matter to argue the point :)

Comment (kuasha420) [reply assets/js/modules/analytics/components/settings/GA4SettingsControls.js]: Well, I had to extract out the error related logic to a separate component anyway, as there were a few more changes that came from other merged PRs that put the complexity to the roof. Hence, I have reverted this change here but made @tofumatt's life harder by introducing another component and corresponding tests :sweat_smile:

File: assets/js/modules/analytics/components/settings/PropertyOrWebDataStreamNotAvailableError.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/settings/PropertyOrWebDataStreamNotAvailableError.js:119]:
```suggestion
	if ( ! propertyAvailable && getPropertiesError === false ) {
```

Comment (tofumatt) [reply assets/js/modules/analytics/components/settings/PropertyOrWebDataStreamNotAvailableError.js:119]: Ah, this is what was causing the bug I encountered: `getPropertiesError` can be `undefined` (for instance, while loading) and it causes a flag of this message when no property is available and there's no error.

Changing this to `=== false` fixes the issue.

**Review by tofumatt — APPROVED** [review]

File: assets/js/modules/analytics/components/settings/PropertyOrWebDataStreamNotAvailableError.js
Comment (tofumatt) [reply assets/js/modules/analytics/components/settings/PropertyOrWebDataStreamNotAvailableError.js]: I was wrong about that, the issue is actually that the previously-selected property for an old account isn't reset until the new properties are loaded, meaning there's a time when the property ID is valid while things are still loading.

The trick should be "resetting" the property ID to be `undefined` when the account changes.

**Comment by tofumatt** [comment]
I needed to make this change: 52031298028f631af4ef9a1bdc3bec11eb9f650c to get everything working and prevent this error: https://github.com/google/site-kit-wp/pull/6816#pullrequestreview-1385972931.

I've otherwise code-reviewed this PR and it looks good, but I'd appreciate a quick MR to make sure this won't cause any problems I haven't noticed. In my testing with GA4 Reporting feature flag on and off, and settings for all combinations of both/either modules, it worked fine.

But it'd be good to get another set of eyes on just this commit: 52031298028f631af4ef9a1bdc3bec11eb9f650c

**Comment by kuasha420** [comment]
@tofumatt Thank you for spotting the edge case, figuring out a fix for it and getting this merged. :tada:

File: assets/js/components/notifications/BannerNotifications.js
Comment (aaemnnosttv) [reply assets/js/components/notifications/BannerNotifications.js:91]: ⚠️ This placement seems incorrect because the component doesn't depend on GTE or the GTM scope. It should however be only shown to authenticated users since it uses module settings which won't be available in a view-only context.

Comment (aaemnnosttv) [reply assets/js/components/notifications/BannerNotifications.js:91]: After further review, it seems okay since the code that triggers it only happens when syncing Google tag settings so while this seems off at first glance, it should be fine.

**Pull request merged**
